rust/agents/validator/src/validator.rs

```diff
@@ -58,12 +58,10 @@ impl BaseAgent for Validator {
         let db = DB::from_path(&settings.db)?;
         let msg_db = HyperlaneRocksDB::new(&settings.origin_chain, db);
 
-        let signer = settings
-            .validator
-            // Intentionally using hyperlane_ethereum for the validator's signer
-            .build::<hyperlane_ethereum::Signers>()
-            .await
-            .map(|validator| Arc::new(validator) as Arc<dyn HyperlaneSigner>)?;
+        // Intentionally using hyperlane_ethereum for the validator's signer
+        let (signer_instance, signer_handle) =
```

Comment:
I was thinking about having the signer just return the handle when created and run itself. Still could do that. Only reason I didn't is I think we want it to be run by the same run all as the agent with a span set there. If not, it would definitely be cleaner to just run it and only return the handle.